com.unity.render-pipelines.core/Editor/Utilities/MenuUtilities.cs

```diff
@@ -0,0 +1,159 @@
+                    {
+                        menuItemsForRP = new List<MenuItemData>();
+                        s_MenuMap[rpAssetType] = menuItemsForRP;
+                    }
+
+                    var query = menuItemsForRP.Where(mi => mi.attribute.path == attribute.path);
+                    bool alreadyHere = query.Count() > 0;
+                    var menuItemData = alreadyHere ? query.First() : new MenuItemData();
+
+                    menuItemData.attribute = attribute;
+                    if (attribute.validate)
+                        menuItemData.validate = (Func<bool>)Delegate.CreateDelegate(typeof(Func<bool>), methodInfo);
+                    else
+                        menuItemData.action = (Action)Delegate.CreateDelegate(typeof(Action), methodInfo);
+
+                    if (!alreadyHere)
+                        menuItemsForRP.Add(menuItemData);
+                }
+            }
+
+            //activating menuitems
+            UpdateMenuItemForRenderPipeline();
+        }
+
+        static void UpdateMenuItemForRenderPipeline()
+        {
+            // remove
+            foreach (var menuItemData in s_MenuMap.Values.SelectMany(mid => mid))
+                UnregisterMenu(menuItemData); //do nothing if not registered before
+
+            // add if required
+
+            // Note: GraphicsSettings.currentRenderPipeline == null possible and should mean built-in
+            // though we cannot use null as key so it was early replaced by typeof(RenderPipelineAsset)
+            Type currentSRPType = GraphicsSettings.currentRenderPipeline?.GetType() ?? typeof(RenderPipelineAsset);
+            if (!s_MenuMap.TryGetValue(currentSRPType, out var menuItemDatasForRP))
+                return;
+
+            foreach (var menuItemData in menuItemDatasForRP)
+                RegisterMenu(menuItemData);
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Method)]
+    public class MenuItemForRenderPipeline : Attribute
+    {
+        public readonly Type[] renderPipelineAssetTypes;
+        public readonly string path;
+        public readonly int priority;
+        public readonly bool validate;
+
+        public MenuItemForRenderPipeline(string path, bool validate = false, int priority = 0, params Type[] renderPipelineAssetTypes)
+        {
+            if (renderPipelineAssetTypes == null || renderPipelineAssetTypes.Length == 0)
+                throw new ArgumentNullException("Argument cannot be null. Use MenuItem instead if you don't need to filter per RenderPipeline.", "renderPipelineTypes");
+
+            // Note: rpAssetType == null possible and should mean built-in
+            // but this is not possible as a key in s_MenuMap so we use the forbiden RenderPipelineAsset type for that
+            foreach (Type type in renderPipelineAssetTypes)
+                if (!typeof(RenderPipelineAsset).IsAssignableFrom(type) && type != null)
```

Comment:
```suggestion
                if (type != null && !typeof(RenderPipelineAsset).IsAssignableFrom(type))
```
I think is better to check if it is null before